master/buildbot/worker/base.py

```diff
@@ -77,6 +79,8 @@ def checkConfig(self, name, password, max_builds=None,
         @type locks: dictionary
         """
         self.name = name = ascii2unicode(name)
+        self.build_wait_timeout = build_wait_timeout
```

Comment:
Thanks for catching this. It appears that this previously wan't being reconfigured before.